drivers/sensor/odsl8/odsl8.c

```diff
@@ -0,0 +1,130 @@
+#include <drivers/sensor.h>
+#include <drivers/adc.h>
+#include <device.h>
+#include <logging/log.h>
+#include <devicetree.h>
+
+#define DT_DRV_COMPAT leuze_odsl8
+
+LOG_MODULE_REGISTER(LASER, CONFIG_SENSOR_LOG_LEVEL);
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "Laser driver enabled without any devices"
+#endif
+
+struct sensor_config {
+    const struct device *adc;
+    struct adc_channel_cfg channel_config;
+    struct adc_sequence adc_seq;
+    const int meas_resistor;
+};
+
+struct adc_data {
+    uint16_t buffer[1];
+};
+
+static int laser_sample_fetch(const struct device *dev, enum sensor_channel channel) {
+
+    const struct sensor_config *config = dev->config;
+
+    if (!device_is_ready(config->adc)) {
+        LOG_ERR("Device ADC not ready!");
+        return -EINVAL;
+    }
+
+    LOG_INF("start adc reading!");
+    int err = adc_read(config->adc, &config->adc_seq);
+    LOG_INF("finished adc reading!");
+
+    if (err) {
+        LOG_ERR("Failed to read ADC channel (error %d)", err);
+        return err;
+    }
+    return 0;
+
+}
+
+static int laser_channel_get(const struct device *dev,
+                             enum sensor_channel channel,
+                             struct sensor_value *val) {
+
+    if (channel != SENSOR_CHAN_DISTANCE) {
+        val->val1 = 0;
+        val->val2 = 0;
+        LOG_ERR("Wrong channel selection! Sensor only supports channel distance.");
+        return 0;
+    }
+
+    LOG_INF("start channel get!");
+    struct adc_data *data = dev->data;
+    const struct sensor_config *config = dev->config;
+
+    int32_t raw_value = (int32_t) data->buffer[0];
+
+    adc_raw_to_millivolts(adc_ref_internal(config->adc),
+                          config->channel_config.gain,
+                          config->adc_seq.resolution,
+                          &raw_value);
+
+    LOG_INF("finished channel get!");
+    // 4..20mA for 20..50mm (linear)
+    const int slope = (500 - 20) / (20 - 4);
+    int32_t distance = (raw_value * slope / config->meas_resistor - 4 * slope) + 20; // distance in mm
+
+    if (distance < 20 || distance > 500) {
+        LOG_WRN("Sensor not in linear region between 20..500mm!");
+    }
+
+    val->val1 = distance;
+    val->val2 = 0;
```

Comment:
Bitte in metern rausgeben: https://docs.zephyrproject.org/latest/reference/peripherals/sensor.html?highlight=sensor_chan_distance#c.sensor_channel.SENSOR_CHAN_DISTANCE https://docs.zephyrproject.org/latest/reference/peripherals/sensor.html?highlight=sensor_value#c.sensor_value

Reply:
fix:  682dc78644fb6d324bd33cc0ca69095351da2f6b